addons/sale_purchase/tests/test_sale_purchase.py

```diff
@@ -255,3 +255,72 @@ def test_update_ordered_sale_quantity(self):
         self.assertEqual(len(purchase_order2.order_line), 1, "The 2nd PO has only one line")
         self.assertEqual(purchase_line2.sale_line_id, self.sol1_service_purchase_1, "The 2nd PO line came from the SO line sol1_service_purchase_1")
         self.assertEqual(purchase_line2.product_qty, delta, "The quantity of the new PO line is the quantity added on the Sale Line, after first PO confirmation")
+
+    def test_choose_supplier(self):
+        """ Choose supplier based on the ordered quantity and minimum price"""
+        product_to_buy = self.env['product.product'].create({
+            'name': "Furniture Service",
```

Comment:
reduce the args to the required fields and fields necessary for the test -> readability